common/components/common/projects/ApproveGroupsSection.jsx

```diff
@@ -0,0 +1,97 @@
+// @flow
+
+import React from 'react';
+import Button from 'react-bootstrap/Button';
+import ProjectAPIUtils,{ProjectDetailsAPIData} from '../../utils/ProjectAPIUtils.js'
+import CurrentUser from "../../utils/CurrentUser.js";
+import type {MyGroupData} from "../../stores/MyGroupsStore.js";
+import url from "../../utils/url.js";
+import Section from "../../enums/Section.js";
+import _ from "lodash";
+
+type Props = {|
+  project: ?ProjectDetailsAPIData,
+|};
+type State = {|
+  pendingGroups: $ReadOnlyArray<MyGroupData>
+|};
+
+/**
+ * For approving/rejecting group invitations
+ */
+
+class ApproveGroupsSection extends React.PureComponent<Props, State> {
+  constructor(props: Props): void {
+    super(props);
+    this.state = {
+      pendingGroups: this.getPendingGroups(props.project)
+    };
+  }
+
+  componentWillReceiveProps(nextProps: Props): void {
+    this.setState({pendingGroups: this.getPendingGroups(nextProps.project)});
+  }
+  
+  getPendingGroups(project: ?ProjectDetailsAPIData): $ReadOnlyArray<MyGroupData> {
+    return project && project.project_groups.filter((group:MyGroupData) => {
+      return !group.relationship_is_approved && CurrentUser.isCoOwnerOrOwner(project);
+    });
+  }
+  
+  removeGroupFromList(group: MyGroupData): void {
+    const groups: $ReadOnlyArray<MyGroupData> = _.remove(this.state.pendingGroups, (g: MyGroupData) => g === group);
+    this.setState({pendingGroups: groups});
+  }
+  
+  handleClick(group: MyGroupData, approved: boolean): void {
+    // TODO: Add LoadingMessage while processing?
+    // metrics.logGroupInviteProjectSubmitConfirm(this.state.selectedGroup.group_id, this.props.projectId);
```

Comment:
I don't see a trello item to add metrics for this; just noting it so we can make sure to capture the task for later.

Reply:
Added